app/controllers/application_controller/filter/expression.rb

```diff
@@ -52,6 +52,17 @@ def exp_available_cfields # fields on exp_model for check_all, check_any, and ch
       end
     end
 
+    def tags_for_display_filters
```

Comment:
Get this stuf out of the haml/helpers - nice.

FYI, in the long term, I'd like to use `Field` instead of the `MiqExpression#model_details`type code. But that will take time / is for another day.